### DISCUSSION TITLE
Automated Backup Strategy

### DISCUSSION THREAD

**Discussion by ups7** [ghostfolio/ghostfolio#2786] [Q&A]
Hello,

What automated backup strategy do people use with Ghostfolio's data (accounts, transactions etc.)? I'm running it on Synology NAS (with Docker) and wondering what's the simplest option to make a backup of all the data periodically with this kind of a setup. Something which can then be relatively easily restored in case of failure.

Any ideas/thoughts welcome.

Thanks!

**Comment by Avendo07** [comment]
I'd say either backup the directory where the database is mapped to.
Or just export the activities each month manually(or automate it, if there's an option using API).

**Comment by swissbuechi** [comment]
I use a local cronjob with a script to achieve sql backups.

# Setup

## Create backup script

Create file: `sudo nano /opt/ghostfolio/backup.sh`

`backup.sh` content:

```shell
#!/bin/bash

# vars
BACKUP_DIR=/opt/ghostfolio/backup
NUM_BACKUPS_TO_KEEP=10

# create backup directory
mkdir -p ${BACKUP_DIR}

# create backup
docker exec -it ghostfolio-postgres-1 pg_dump --clean -U user -d ghostfolio-db > ${BACKUP_DIR}/dump_`date +%d-%m-%Y"_"%H_%M_%S`.sql

# delete old backups
ls -t ${BACKUP_DIR}/dump_*.sql | tail -n +$((${NUM_BACKUPS_TO_KEEP} + 1)) | xargs -I {} rm {}
```

Add execution permission: `sudo chmod +x /opt/ghostfolio/backup.sh`

Test backup: `sudo /opt/ghostfolio/backup.sh`

## Create cronjob

`sudo crontab -e`

add line: `0 23 * * * /opt/ghostfolio/backup.sh > /opt/ghostfolio/backup.log 2>&1`

# Restore

`cat /opt/ghostfolio/dump_<date>.sql | docker exec -i ghostfolio-postgres-1 psql -U user -d ghostfolio-db`

**Reply by gacolitti** [reply]
This is awesome. I'm trying to use this method to restore data on a new host. But running the restore command seems to render the security token invalid. I've been unable to successfully restore the data from one instance of ghostfolio to another as a result.

**Reply by gacolitti** [reply]
Following [this advice](https://github.com/ghostfolio/ghostfolio/discussions/1653#discussioncomment-4868872) I was able to restore data on a new host.
